Source/ProjectRimFactory/SAL3/Things/Assemblers/Building_ProgrammableAssembler.cs

```diff
@@ -549,6 +549,35 @@ public List<Thing> GetThingQueue()
             return thingQueue;
         }
 
+        public override void DeSpawn(DestroyMode mode = DestroyMode.Vanish)
+        {
+            for (int i = thingQueue.Count - 1; i >= 0; i--)
+            {
+                //some of the things contaned are marked as Destroyed for some reason
+                if (thingQueue[i].Destroyed) thingQueue[i].ForceSetStateToUnspawned();
+
+                PlaceThingUtility.PRFTryPlaceThing(this, thingQueue[i], this.Position, this.Map, true);

```

Comment:
Why do you still place something if it's destroyed?

Reply:
The Idea was that if your assembler with X items in the Queue gets destroyed by lets say:
Insects, Mortar or Gunfire then `public override void Destroy` is not called instead `DeSpawn` is called.
there all items in the Queue except the first one are `Destroyed` _(not sure why)_

I think it should drop those items instead of letting them vanish.
_(We could also let the items take damage if this is a balance concern)_

Reply:
Ah; thank you!